server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -606,7 +735,12 @@ private void constructStarTree(InMemoryTreeNode node, int startDocId, int endDoc
 
         // Construct star-node if required
         if (!skipStarNodeCreationForDimensions.contains(childDimensionId) && children.size() > 1) {
-            children.put((long) ALL, constructStarNode(startDocId, endDocId, childDimensionId));
+            node.childStarNode = constructStarNode(startDocId, endDocId, childDimensionId);
+        }
+
+        // Further split star node if needed
+        if (node.childStarNode != null && (node.childStarNode.endDocId - node.childStarNode.startDocId > maxLeafDocuments)) {
```

Comment:
earlier before this change , `constructStarTree` for star node was called at end after all children, now its reversed.

Can we keep the same logic - call this after we call construct star tree for children.

Reply:
my bad , this is correct

Reply:
> constructStarTree for star node was called at end after all children, now its reversed.

How can we say it is reversed? We had a hashmap (based on dimension value) to store all the children (including star node). Now, hashmap's bucketing algorithm would be deciding which node is picked first.

Star Node is still constructed after the other nodes, as the logic should be, but the possible construction of star node's children should not be affected by other nodes.

Reply:
its not predictable i agree, lets think through maybe after linkedhashmap

Reply:
replaced with linkedhashmap